### PR TITLE
Improve AudioStreamGenerator and AudioEffectSpectrumAnalyzer documentation

### PR DESCRIPTION
**Pull request by Calinou**
Follow-up to https://github.com/godotengine/godot/pull/48681.

- Mention audio sample rate caveats in other classes where relevant.

cc @unfa as he works with audio :slightly_smiling_face:

### CODE REVIEW

File: doc/classes/AudioEffectSpectrumAnalyzer.xml [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
 	</methods>
 	<members>
 		<member name="buffer_length" type="float" setter="set_buffer_length" getter="get_buffer_length" default="2.0">
+			The length of the buffer to keep (in seconds). Higher values keep data around for longer, but require more memory.
 		</member>
 		<member name="fft_size" type="int" setter="set_fft_size" getter="get_fft_size" enum="AudioEffectSpectrumAnalyzer.FFT_Size" default="2">
+			The size of the [url=https://en.wikipedia.org/wiki/Fast_Fourier_transform]Fast Fourier transform[/url] buffer. Higher values result in better quality, but are more demanding on the CPU and may cause audio cracking if the CPU can't keep up.
 		</member>
 		<member name="tap_back_pos" type="float" setter="set_tap_back_pos" getter="get_tap_back_pos" default="0.01">
 		</member>
```
Comment (Calinou) [inline doc/classes/AudioEffectSpectrumAnalyzer.xml:24]: Does anyone know what this property is used for? I'd like to get 100% completion :slightly_smiling_face:

File: doc/classes/AudioStreamGeneratorPlayback.xml [hunk 2]
```diff
@@ -14,18 +17,21 @@
...
 			</description>
 		</method>
 		<method name="get_frames_available" qualifiers="const">
 			<return type="int">
 			</return>
 			<description>
+				Returns the number of audio data frames left to play. If this returned number reaches [code]0[/code], the audio will stop playing until frames are added again. Therefore, make sure your script can always generate and push new audio frames fast enough to avoid audio cracking.
 			</description>
 		</method>
 		<method name="get_skips" qualifiers="const">
```
Comment (Calinou) [inline doc/classes/AudioStreamGeneratorPlayback.xml:37]: Does anyone know what this method is used for? I'd like to get 100% completion :slightly_smiling_face:

File: doc/classes/AudioEffectSpectrumAnalyzer.xml [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
+		<link title="https://godotengine.org/article/godot-32-will-get-new-audio-features">Godot 3.2 will get new audio features</link>
 	</tutorials>
 	<methods>
 	</methods>
 	<members>
 		<member name="buffer_length" type="float" setter="set_buffer_length" getter="get_buffer_length" default="2.0">
+			The length of the buffer to keep (in seconds). Higher values keep data around for longer, but require more memory.
 		</member>
 		<member name="fft_size" type="int" setter="set_fft_size" getter="get_fft_size" enum="AudioEffectSpectrumAnalyzer.FFT_Size" default="2">
+			The size of the [url=https://en.wikipedia.org/wiki/Fast_Fourier_transform]Fast Fourier transform[/url] buffer. Higher values result in better quality, but are more demanding on the CPU and may cause audio cracking if the CPU can't keep up.
```
Comment (Calinou) [inline doc/classes/AudioEffectSpectrumAnalyzer.xml:21]: Does the FFT size actually affect the visualization quality, or just the history size available? Does it have any relationship with the `buffer_length`?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Pull request merged**